spec/ddtrace/configuration/agent_settings_resolver_spec.rb

```diff
@@ -16,22 +16,61 @@
   let(:ddtrace_settings) { Datadog::Configuration::Settings.new }
   let(:logger) { instance_double(Datadog::Logger) }
 
-  let(:default_settings) do
+  let(:settings) do
     {
+      adapter: adapter,
       ssl: false,
-      hostname: '127.0.0.1',
-      port: 8126,
+      hostname: hostname,
+      port: port,
+      uds_path: uds_path,
       timeout_seconds: 1,
       deprecated_for_removal_transport_configuration_proc: nil,
-      deprecated_for_removal_transport_configuration_options: nil
+      deprecated_for_removal_transport_configuration_options: nil,
     }
   end
```

Comment:
I can live with, but I don't quite like this change. The specs below all had the pattern of "when you do this, then the outcome is default settings + your change". With this change, it means that the settings may have been changed at another level, which I think is a bit less readable.

My alternative suggestion would be to have a `uds_settings` let and use that in the uds tests.